java/clients/src/test/resources/reports/filestatus/standard/splitFile.txt

```diff
@@ -0,0 +1,20 @@
+
+Files Status Report:
+--------------------------
+There are 2 leaf partitions and 1 non-leaf partitions
+There are 1 files with status of "Ready_to_be_garbage_collected"
+	(0 in leaf partitions, 1 in non-leaf partitions)
+There are 4 files with status of "Active"
+	(3 in leaf partitions, 1 in non-leaf partitions)
+Number of files in leaf partitions: min = 1, max = 2, average = 1.5
+Number of files in non-leaf partitions: min = 1, max = 1, average = 1.0
+Ready_to_be_garbage_collected:
+FileInfo{filename='split.parquet', partitionId='A', numberOfRecords=2000, fileStatus=READY_FOR_GARBAGE_COLLECTION, jobId='null', lastStateStoreUpdateTime=1661178000001, countApproximate=false, onlyContainsDataForThisPartition=true}
+Active:
+FileInfo{filename='not-split.parquet', partitionId='A', numberOfRecords=1000, fileStatus=ACTIVE, jobId='null', lastStateStoreUpdateTime=1661178000001, countApproximate=false, onlyContainsDataForThisPartition=true}
+FileInfo{filename='pending-split.parquet', partitionId='B', numberOfRecords=2000, fileStatus=ACTIVE, jobId='null', lastStateStoreUpdateTime=1661178000001, countApproximate=false, onlyContainsDataForThisPartition=true}
+FileInfo{filename='split-1.parquet', partitionId='B', numberOfRecords=1000, fileStatus=ACTIVE, jobId='null', lastStateStoreUpdateTime=1661178000001, countApproximate=true, onlyContainsDataForThisPartition=false}
+FileInfo{filename='split-2.parquet', partitionId='C', numberOfRecords=1000, fileStatus=ACTIVE, jobId='null', lastStateStoreUpdateTime=1661178000001, countApproximate=true, onlyContainsDataForThisPartition=false}
+Total number of records in all active files (approx) = 5K (5,000)
+Total number of records in leaf partitions (approx) = 4K (4,000)
+Percentage of records in leaf partitions (approx) = 80.0
```

Comment:
I feel like the distinction of "in leaf partitions" is a bit confusing once we've got the new splitting compactions working. The only way the number of records in all files would be different from the number of records in leaf partitions is if the metadata splits haven't happened yet. We might decide to do those straight away on ingest or partition splitting, in which case those numbers should never be different.

It's still useful to know when a file is not associated with any leaf partitions, but it seems like it might be more important to know how many of the records are estimated vs known exactly?

The issue says "The total records at the bottom of the report should include a more detailed count of which records are in split files and which records are in files that have been compacted." Right now there's no summary of which records are in split files and which are in whole files.

Maybe it could be something like:

```
Total number of records in all active files (approx): 5K (5,000)
Total number of records indexed by leaf partition: 1K (1,000)
Total number of records indexed by non-leaf partition: 2K (2,000)
Total number of records estimated in leaf partitions: 2K (2,000)
Total number of records indexed by any partition: 3K (3,000)
```

Reply:
it's true that in time there may hardly ever be any records that are not in leaf partitions as we might just move them down straight away (I think there will always be a chance of there being some in non-leaf partitions as if an ingest job finishes, commits, and then a partition is split at the same time, then there will be a delay before we move the file down). I think the terminology "indexed by" is confusing. In your example shouldn't "Total number of records indexed by leaf partition" and "Total number of records indexed by non-leaf partition" add up to 5000?

I'm tempted to leave it as it is for now and improve it later if needed.

Reply:
I imagined "indexed by" would mean the sketches and the index in the parquet file are fully aligned with the partition that the FileInfo record is for. In that case, the example has 3K records which are indexed against the partition (1K against a leaf and 2K against a non-leaf), and 2K which are indexed against an unknown partition which is different from the partition on the FileInfo record.